kernel/pci/src/lib.rs

```diff
@@ -211,84 +249,133 @@ pub struct PciLocation {
     func: u8,
 }
 
+struct PciAddress {
+    /// Offset to a double word in a PCI configuration space
+    dword_address: u32,
```

Comment:
this isn't really an address, it's an offset into the configuration space.

Also, if you adopt the redesign suggestion below, it would be an index into the slice of volatile `u32`s, so that might help with the naming convention and documentation clarity here.

Reply:
> this isn't really an address, it's an offset into the configuration space.

I was re-using the original terminology of the crate but I agree

Reply:
Now that we represent the configuration space as an array of `u32`s, let's just call this what it is -- an index.